code/datums/gamemode/factions/bloodcult/bloodcult_projectiles.dm

```diff
@@ -0,0 +1,284 @@
+//                          //Used when a filled soul blade performs a perforation
+//////////////////////////////
+
+/obj/item/projectile/soulbullet
+	name = "soul blade"
+	icon = 'icons/obj/cult_64x64.dmi'
+	icon_state = "soulbullet"
+	pixel_x = -16 * PIXEL_MULTIPLIER
+	pixel_y = -10 * PIXEL_MULTIPLIER
+	damage = 30//Only affects obj/turf. Mobs take a regular hit from the sword.
+	phase_type = PROJREACT_MOBS
+	penetration = -1
+	fire_sound = null
+	mouse_opacity = 1
+	var/turf/secondary_target = null
+	var/obj/item/weapon/melee/soulblade/blade = null
+	var/mob/living/simple_animal/shade/shade = null
+	var/redirected = 0
+	var/leave_shadows = -1
+	var/matrix/shadow_matrix = null
+
+/obj/item/projectile/soulbullet/Destroy()
+	var/turf/T = get_turf(src)
+	if (T)
+		if (blade)
+			blade.forceMove(T)
+	blade = null
+	shade = null
+	..()
+
+/obj/item/projectile/soulbullet/OnFired(var/proj_target = original)
+	target = get_turf(proj_target)
+	if (!secondary_target)
+		secondary_target = target
+	if (!shade)
+		icon_state = "soulbullet-empty"
+	dist_x = abs(target.x - starting.x)
+	dist_y = abs(target.y - starting.y)
+	override_starting_X = starting.x
+	override_starting_Y = starting.y
+	override_target_X = target.x
+	override_target_Y = target.y
+	if (target.x > starting.x)
+		dx = EAST
+	else
+		dx = WEST
+	if (target.y > starting.y)
+		dy = NORTH
+	else
+		dy = SOUTH
+	if(dist_x > dist_y)
+		error = dist_x/2 - dist_y
+	else
+		error = dist_y/2 - dist_x
+	if (target!=secondary_target)
+		target_angle = round(Get_Angle(target,secondary_target))
+		blade.dir = get_dir(target,secondary_target)
+	else
+		target_angle = round(Get_Angle(starting,target))
+		blade.dir = get_dir(starting,target)
+	shadow_matrix = turn(matrix(),target_angle+45)
+	transform = shadow_matrix
+	//var/matrix/base_matrix = turn(matrix(),target_angle)
+	//var/image/I = image('icons/obj/cult_64x64.dmi',"[icon_state]_spin")
+	//I.transform = base_matrix
+	if (shade)
+		icon_state = "soulbullet_spin"
+		plane = HUD_PLANE
+		layer = ABOVE_HUD_LAYER
+	else
+		icon_state = "soulbullet-empty_spin"
+	spawn(5)
+		leave_shadows = 0
+		/*
+		if( !("[icon_state]_angle[target_angle]" in bullet_master) )
+			var/image/I = new('icons/obj/cult_64x64.dmi',"[icon_state]")
+			//I.transform = base_matrix
+			bullet_master["[icon_state]_angle[target_angle]"] = I
+		src.icon = bullet_master["[icon_state]_angle[target_angle]"]
+		*/
+		if (shade)
+			icon_state = "soulbullet"
+		else
+			icon_state = "soulbullet-empty"
+	return 1
+
+/obj/item/projectile/soulbullet/bresenham_step(var/distA, var/distB, var/dA, var/dB)
+	if (shade && leave_shadows >= 0)
+		leave_shadows++
+		if ((leave_shadows%3)==0)
+			anim(target = loc, a_icon = 'icons/obj/cult_64x64.dmi', flick_anim = "soulblade-shadow", lay = NARSIE_GLOW, offX = pixel_x, offY = pixel_y, plane = LIGHTING_PLANE, trans = shadow_matrix)
+	if(..())
+		return 2
```

Comment:
Magic number

Reply:
I'll argue that it's not a variable that's meant to be either changed, nor used anywhere else, so there's little benefit to have it be a define, unlike let's say tattoo names.